### PR TITLE
feat: improve typings

### PR DESCRIPTION
**Pull request by splincode**
(cherry picked from commit b250d5d2a64ba2f0e21fc0625cd2e1e6ed860b38)

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

### CODE REVIEW

File: projects/ng-polymorpheus/src/classes/component.ts [hunk 1]
```diff
@@ -7,7 +7,10 @@ import {POLYMORPHEUS_CONTEXT} from '../tokens/context';
  * @param component — an Angular component to be dynamically created
  * @param injector — optional {@link Injector} for lazy loaded module case
  */
-export class PolymorpheusComponent<T extends object, C extends object> {
+export class PolymorpheusComponent<
+    T extends Record<any, any>,
```
Comment (waterplea) [inline projects/ng-polymorpheus/src/classes/component.ts:11]: T is a constructor for the component, I think we can drop `extends` from it completely and just leave it as `T`

File: projects/ng-polymorpheus/src/directives/outlet.ts [hunk 2]
```diff
@@ -99,20 +100,20 @@ export class PolymorpheusOutletDirective<C extends object> implements OnChanges,
...
 ): content is PolymorpheusTemplate<C> {
     return content instanceof PolymorpheusTemplate;
 }
 
-function isComponent<C extends object>(
-    content: PolymorpheusContent<C> | null,
-): content is PolymorpheusComponent<object, C> {
+function isComponent<C extends Record<any, any>>(
+    content: PolymorpheusContent<C>,
+): content is PolymorpheusComponent<Record<any, any>, C> {
```
Comment (waterplea) [inline projects/ng-polymorpheus/src/directives/outlet.ts:111]: Just `any` here, I guess. Or maybe unknown if it doesn't break anything.

File: projects/ng-polymorpheus/src/tokens/context.ts [hunk 3]
```diff
@@ -4,6 +4,6 @@ import {InjectionToken} from '@angular/core';
  * Use this token to access context within your components when
  * instantiating them through {@link PolymorpheusOutletDirective}
  */
-export const POLYMORPHEUS_CONTEXT = new InjectionToken<object>(
+export const POLYMORPHEUS_CONTEXT = new InjectionToken<unknown>(
```
Comment (waterplea) [inline projects/ng-polymorpheus/src/tokens/context.ts:7]: Context should be `Record<any, any>`

File: projects/ng-polymorpheus/src/types/content.ts [hunk 4]
```diff
@@ -2,14 +2,16 @@ import {TemplateRef} from '@angular/core';
...
 
 /**
  * All content types supported by {@link PolymorpheusOutletDirective}
  */
-export type PolymorpheusContent<C extends object = {}> =
+export type PolymorpheusContent<C extends Record<any, any> = {}> =
     | TemplateRef<C>
     | PolymorpheusTemplate<C>
-    | PolymorpheusComponent<object, C>
+    | PolymorpheusComponent<Record<any, any>, C>
```
Comment (waterplea) [inline projects/ng-polymorpheus/src/types/content.ts:12]: Probably `any`, I think `unknown` might not work.

File: projects/ng-polymorpheus/src/classes/component.ts (hunk 1 above)
Comment (splincode) [reply projects/ng-polymorpheus/src/classes/component.ts]: fixed

File: projects/ng-polymorpheus/src/tokens/context.ts (hunk 3 above)
Comment (splincode) [reply projects/ng-polymorpheus/src/tokens/context.ts]: fixed

File: projects/ng-polymorpheus/src/types/content.ts (hunk 4 above)
Comment (splincode) [reply projects/ng-polymorpheus/src/types/content.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**